ngraph/test/backend/convert.in.cpp

```diff
@@ -361,31 +432,102 @@ NGRAPH_TEST(${BACKEND_NAME}, convert_u8_to_i64)
+{
+    const std::vector<uint8_t> input{0x10, 0x01, 0x00, 0x00};
+    const Shape input_shape{8};
+    const element::Type input_type = ngraph::element::u4;
+
+    const std::vector<uint8_t> expected_output{0x90};
+    const element::Type expected_output_type = ngraph::element::u1;
+
+    LPConvertTest(input, input_shape, input_type, expected_output, expected_output_type);
+}
+
+NGRAPH_TEST(${BACKEND_NAME}, convert_u8_to_u1)
+{
+    const std::vector<uint8_t> input{1, 0, 1, 0, 0, 0, 0, 1};
+    const Shape input_shape{8};
     const element::Type input_type = ngraph::element::u8;
 
-    const std::vector<uint8_t> expected_output{0x00};
+    const std::vector<uint8_t> expected_output{0xA1};
     const element::Type expected_output_type = ngraph::element::u1;
 
-    ASSERT_THROW(ConvertTest(input, input_shape, input_type, expected_output, expected_output_type),
-                 ngraph::NodeValidationFailure);
+    LPConvertTest(input, input_shape, input_type, expected_output, expected_output_type);
+}
+
+NGRAPH_TEST(${BACKEND_NAME}, convert_i4_to_u1)
+{
+    const std::vector<uint8_t> input{0x10, 0x01, 0x00, 0x00};
+    const Shape input_shape{8};
+    const element::Type input_type = ngraph::element::u4;
+
+    const std::vector<uint8_t> expected_output{0x90};
+    const element::Type expected_output_type = ngraph::element::u1;
+
+    LPConvertTest(input, input_shape, input_type, expected_output, expected_output_type);
 }
 
 // destination: u4
-NGRAPH_TEST(${BACKEND_NAME}, convert_u8_to_u4_is_not_supported_yet)
+NGRAPH_TEST(${BACKEND_NAME}, convert_u1_to_u4)
 {
-    const std::vector<uint8_t> input{0, 0, 0, 0};
+    const std::vector<uint8_t> input{0xF0};
+    const Shape input_shape{4};
+    const element::Type input_type = ngraph::element::u1;
+
+    const std::vector<uint8_t> expected_output{0x11, 0x11};
+    const element::Type expected_output_type = ngraph::element::u4;
+
+    LPConvertTest(input, input_shape, input_type, expected_output, expected_output_type);
+}
+
+NGRAPH_TEST(${BACKEND_NAME}, convert_u4_to_u4)
+{
+    const std::vector<uint8_t> input{0x22, 0x33};
+    const Shape input_shape{4};
+    const element::Type input_type = ngraph::element::u4;
+
+    const std::vector<uint8_t> expected_output{0x22, 0x33};
+    const element::Type expected_output_type = ngraph::element::u4;
+
+    LPConvertTest(input, input_shape, input_type, expected_output, expected_output_type);
+}
+
+
+NGRAPH_TEST(${BACKEND_NAME}, convert_u8_to_u4)
+{
+    const std::vector<uint8_t> input{7, 0, 1, 15};
     const Shape input_shape{4};
     const element::Type input_type = ngraph::element::u8;
 
-    const std::vector<uint8_t> expected_output{0x00, 0x00};
+    const std::vector<uint8_t> expected_output{0x70, 0x1F};
+    const element::Type expected_output_type = ngraph::element::u4;
+
+    LPConvertTest(input, input_shape, input_type, expected_output, expected_output_type);
+}
+
+NGRAPH_TEST(${BACKEND_NAME}, convert_i4_to_u4)
```

Comment:
Can we add tests to convert {lp types} -> {C++ int types} and {C++ int types} -> {lp types}?

Reply:
Sure, since you've merge it already I'll prepare next PR today with it. Thx Ilya!

Reply:
Test will be introduced in #5745.